### ISSUE TITLE
`IndicatorResult` need refactoring

### ISSUE THREAD

**Issue by amv-dev** [amv-dev/yata#13]
For now every indicator must implement both `value`s and `signal`s calculation logic. That might be a problem if you want to use values to implement your own `signal`s logic because of performance degradation.

So the current proposal is to make an `IndicatorResult` as a `trait` which will only be responsible for representing values. The idea behind is that every indicator should have it's own `IndicatorResult` type. This will help keep indicators values more verbose.
A single `struct`, that implements `IndicatorResult`, must contain all the values, that indicator returns.

Then for signals create another `trait` `Signal` and make to signal types: `BinarySignal` for signals like {-1; 0; +1} and `FloatSignal` for signals in range `[-1.0; 1.0]`. Every single signal must have it's own `struct` representation.

**Comment by raimundomartins** [comment]
I agree with this. Personally, I don't care about Keltner Channel's signals, but want to use its values.

I'm also not a fan of the (arbitrary) 4 signals limit nor the `enum Action`, which feels it can be completely replaced by a `f32` and would be improved. Custom structs for indicators are a definite improvement!
